From 08e04c01fd752b8d71745803e625ffd3f7ddf0bb Mon Sep 17 00:00:00 2001
From: Alan Modra <amodra@gmail.com>
Date: Fri, 8 Feb 2019 11:51:34 +1030
Subject: [PATCH 014/128] Make inline plt reloc "unsupported for bss-plt" an
 error

This was always supposed to be an error.  Code emitted by gcc for
inline PLT calls assumes PLT is an array of addresses.

	* elf32-ppc.c (ppc_elf_relocate_section): Add %X to "unsupported
	for bss-plt" warning to make it an error.

(cherry picked from commit 482f3505d1b62cbcf46ffed54807fad0d91c8f09)
---
 bfd/ChangeLog   | 5 +++++
 bfd/elf32-ppc.c | 4 ++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/bfd/elf32-ppc.c b/bfd/elf32-ppc.c
index 10c04b997f..593ef47592 100644
--- a/bfd/elf32-ppc.c
+++ b/bfd/elf32-ppc.c
@@ -8852,7 +8852,7 @@ ppc_elf_relocate_section (bfd *output_bfd,
 	    }
 	  else if (htab->plt_type != PLT_NEW)
 	    info->callbacks->einfo
-	      (_("%P: %H: %s relocation unsupported for bss-plt\n"),
+	      (_("%X%P: %H: %s relocation unsupported for bss-plt\n"),
 	       input_bfd, input_section, rel->r_offset,
 	       howto->name);
 	  break;
@@ -8870,7 +8870,7 @@ ppc_elf_relocate_section (bfd *output_bfd,
 	    }
 	  else if (htab->plt_type != PLT_NEW)
 	    info->callbacks->einfo
-	      (_("%P: %H: %s relocation unsupported for bss-plt\n"),
+	      (_("%X%P: %H: %s relocation unsupported for bss-plt\n"),
 	       input_bfd, input_section, rel->r_offset,
 	       howto->name);
 	  break;
-- 
2.21.0

